tests/cache.rs

```diff
@@ -10,7 +13,7 @@ async fn test_bundle_poller_roundtrip() -> eyre::Result<()> {
     let config = setup_test_config().unwrap();
 
     let (block_env, _jh) = config.env_task().spawn();
-    let cache = config.spawn_cache_system(block_env);
+    let cache = CacheSystem::spawn(&config, block_env);
```

Comment:
this API doesn't match our idiom. our idiom is to 

- instantiate a struct `let my_struct = MyStruct::new(...)`
- call `my_struct.spawn()`

`CacheSystem` is not a spawnable task in this way, it;s only a holder for the join handles and a ref to the simcache to ensure it is not prematurely dropped

if you want to make a spawnable `CacheTasks` then it should be a composition of the 2 component tasks that spawns both of them

```
struct CacheTask {
    /// The transaction poller task.
    pub tx: TxPoller,

    /// The bundle poller task.
    pub bundle: BundlePoller
}
```